tempesta_fw/sock_srv.c

```diff
@@ -100,17 +100,17 @@ typedef struct {
 } TfwSrvConnection;
 
 /**
- * Initiate a new connect attempt without blocking
- * until the connection is established.
+ * Initiate a non-blocking connect attempt.
+ * Returns immediately without waiting until a connection is established.
  */
 static int
-tfw_sock_srv_connect_try(TfwSrvConnection *srv_conn)
+tfw_sock_srv_connect_try(TfwConnection *conn)
```

Comment:
Changes in whole patch https://github.com/natsys/tempesta/commit/c5a198fc8d3795e03d1c43311ba20c6ac8e81c2c are doubtful. It's somewhat like broken incapsulation. sock_srv.c is a set of functions working with TfwSrvConnection, a derived class from TfwConnection. So the static functions of sock_srv.c a likely working with various members of TfwSrvConnection (probably some of them in future rather than now), so it's better to keep them like static methods of a class (in C++ terms, TfwSrvConnection in our case) accepting the class as the first argument. However, external interfaces of the class (e.g. tfw_srv_conn_free()) are really could be called with TfwConnection. 

Thus generic rule for such cases: external interfaces can work with generic base class, while internal (static) functions should work with derived class.